Authentication/IamTokenManager.cs

```diff
@@ -0,0 +1,186 @@
+            }
+        }
+
+        #region Request Token
+        /// <summary>
+        /// Request an IAM token using an API key.
+        /// </summary>
+        /// <param name="callback">The request callback.</param>
+        /// <param name="error"> The request error.</param>
+        /// <returns></returns>
+        override protected bool RequestToken(Callback<TokenData> callback)
+        {
+            if (callback == null)
+                throw new ArgumentNullException("successCallback");
+
+            RESTConnector connector = new RESTConnector();
+            connector.URL = url;
+            if (connector == null)
+                return false;
+
+            RequestIamTokenRequest req = new RequestIamTokenRequest();
+            req.Callback = callback;
+            req.HttpMethod = UnityWebRequest.kHttpVerbGET;
+            req.Headers.Add("Content-type", "application/x-www-form-urlencoded");
+            req.Headers.Add("Authorization", Utility.CreateAuthorization(iamClientId, iamClientSecret));
+            req.OnResponse = OnRequestIamTokenResponse;
+            req.DisableSslVerification = disableSslVerification;
+            req.Forms = new Dictionary<string, RESTConnector.Form>();
+            req.Forms["grant_type"] = new RESTConnector.Form("urn:ibm:params:oauth:grant-type:apikey");
+            req.Forms["apikey"] = new RESTConnector.Form(iamApikey);
+            req.Forms["response_type"] = new RESTConnector.Form("cloud_iam");
+
+            return connector.Send(req);
+        }
+
+        private class RequestIamTokenRequest : RESTConnector.Request
+        {
+            public Callback<TokenData> Callback { get; set; }
+        }
+
+        private void OnRequestIamTokenResponse(RESTConnector.Request req, RESTConnector.Response resp)
+        {
+            DetailedResponse<TokenData> response = new DetailedResponse<TokenData>();
+            response.Result = new TokenData();
+            foreach (KeyValuePair<string, string> kvp in resp.Headers)
+            {
+                response.Headers.Add(kvp.Key, kvp.Value);
+            }
+            response.StatusCode = resp.HttpResponseCode;
+
+            try
+            {
+                string json = Encoding.UTF8.GetString(resp.Data);
+                response.Result = JsonConvert.DeserializeObject<TokenData>(json);
+                response.Response = json;
+            }
+            catch (Exception e)
+            {
+                Log.Error("Credentials.OnRequestIamTokenResponse()", "Exception: {0}", e.ToString());
+                resp.Success = false;
+            }
+
+            if (((RequestIamTokenRequest)req).Callback != null)
+                ((RequestIamTokenRequest)req).Callback(response, resp.Error);
+        }
+        #endregion
+    }
+
+    public class IamTokenOptions : JwtTokenOptions
+    {
+        private string iamApiKey;
+        [JsonProperty("iamApiKey", NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
Do we need serialize names for these? Are they coming from or converted to json?

Reply:
`IamTokenOptions` has these properties, they are coming in as `json`. We can remove them as we are setting values from options in constructor too.

Reply:
Actually, on thinking about it again I think we should keep these options separate for child token manager classes because we are passing them on to token manager as token options and some of the values are local to Iam token manager or Icp4d token manager.